src/app/modules/main/chat_section/module.nim

```diff
@@ -698,6 +770,27 @@ method onChatMuted*(self: Module, chatId: string) =
 method onChatUnmuted*(self: Module, chatId: string) =
   self.view.chatsModel().changeMutedOnItemById(chatId, muted=false)
 
+method onCommunityTokenPermissionDeleted*(self: Module, communityId: string, permissionId: string) =
+  self.view.tokenPermissionsModel().removeItemWithId(permissionId)
+  singletonInstance.globalEvents.showCommunityTokenPermissionDeletedNotification(communityId, "Community permission deleted", "A token permission has been removed")
+
+method onCommunityTokenPermissionCreated*(self: Module, communityId: string, tokenPermission: CommunityTokenPermissionDto) =
+  self.rebuildCommunityTokenPermissionsModel()
```

Comment:
Can't we just add to the model? Wouldn't we only check if the new criteria is met anyway?

Reply:
So originally I had it indeed implemented such that `onCommunityTokenPermissionCreated()` and `onCommunityTokenPermissionUpdated()` would just add and update an item respectively.

And because we need to revalidate whether we have still meet all token criteria when a permission was updated, I ended up using the rebuildCommunityTokenPermissionsModel() which does the revalidation but also updates the model.

However, now that I'm writing this, I realized there's a different issue with
`rebuildCommunityTokenPermissionsModel()` w.r.t. updateing the `allTokenCriteriaMet` flag...

I'll revisit this.